src/components/Comment/NewCommentForm.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useCallback, useContext, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import classNames from 'classnames';
+import { CommentData } from '../../types/Comment';
+import {
+  DispatchContext,
+  StateContext,
+} from '../../reducer/store';
+import { commentService } from '../../services/comment.service';
+
+type Props = {
+  postId: number
+};
+
+export const NewCommentForm: React.FC<Props> = React.memo(({ postId }) => {
+  const [showSpinner, setShowSpinner] = useState(false);
+
+  const { commentsList } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  const {
+    register,
+    formState: { errors },
+    reset,
+    resetField,
+    handleSubmit,
+  } = useForm<CommentData>();
+
+  const onSubmit = (data: CommentData) => {
+    const { addNewComment } = commentService(dispatch, commentsList);
+
+    setShowSpinner(true);
+
+    const dataToSend = {
+      name: data.name.trim(),
+      email: data.email.trim(),
+      body: data.body.trim(),
+      postId,
+    };
+
+    addNewComment(dataToSend, setShowSpinner, resetField);
```

Comment:
Calling the service directly from the component violates separation of concerns. Consider using an action creator to perform async operations and dispatch the result.